src/Chapter2.hs

```diff
@@ -501,8 +518,23 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+isThird42 :: [Int] -> Bool
+isThird42 (_:_:42:_) = True
+isThird42 _ = False
 
+{-
+  Why here if I am writing 
+    @
+    isThird42 (_:_:42) = True
+    @
+  I am getting an error:
+
+  src/Chapter2.hs:522:16: error:
+    • No instance for (Num [Int]) arising from the literal ‘0’
+    • In the pattern: 0
+      In the pattern: _ : 0
+      In the pattern: _ : _ : 0
+-} 
```

Comment:
When you use pattern `(x : xs)`, the `x` variable has the type of the list element, and `xs` has the type of list. So when you writ e `(x : 42)` the compiler tries to match types of the number 42 and the list and fails (because number type is not the same as the list type)

Reply:
Thank you for explanation!